src/harness.ts

```diff
@@ -0,0 +1,493 @@
+	 */
+	public classes(...classes: (string | null)[]): { [className: string ]: boolean } {
+		const result: { [className: string]: boolean } = {};
+
+		this._classes.reduce((result, className) => {
+			result[className] = false;
+			return result;
+		}, result);
+
+		classes.reduce((result, className) => {
+			if (className) {
+				result[className] = true;
+				if (!includes(this._classes, className)) {
+					this._classes.push(className);
+				}
+			}
+			return result;
+		}, result);
+
+		return result;
+	}
+
+	/**
+	 * Assert an expected virtual DOM (`DNode`) against what is actually being rendered.  Will throw if the expected does
+	 * not match the actual.
+	 * @param expected The expected render (`DNode`)
+	 * @param message Any message to be part of an error that gets thrown if the actual and expected do not match
+	 */
+	public expectRender(expected: DNode, message?: string): this {
+		this._widgetHarness.expectedRender = expected;
+		this._widgetHarness.assertionMessage = message;
+		this._widgetHarness.didRender = false;
+		this._invalidate();
+		if (!this._widgetHarness.didRender) {
+			throw new Error('An expected render did not occur.');
+		}
+		return this;
+	}
+
+	/**
+	 * Refresh the render and return the last render's root `DNode`.
+	 */
+	public getRender(): DNode {
+		this._invalidate();
+		return this._widgetHarness.lastRender!;
+	}
+
+	/**
+	 * Get the root element of the harnessed widget.  This will refresh the render.
+	 */
+	public getDom(): HTMLElement {
+		if (!this._attached) {
+			this._invalidate();
+		}
+		if (!(this._root && this._root.firstChild)) {
+			throw new Error('No root node has been rendered');
+		}
+		return <HTMLElement> this._root.firstChild;
+	}
+
+	/**
+	 * Clear any cached classes that have been cached via calls to `.classes()`
+	 */
+	public resetClasses(): this {
+		this._classes = [];
+		return this;
+	}
+
+	/**
+	 * Dispatch an event to the root DOM element of the rendered harnessed widget.  You can use the options to change the
+	 * event class, provide additional event properties, or select a different target.
+	 *
+	 * By default, the event class is `CustomEvent` and `bubbles` and `cancelable` are both `true` on events dispatched by
+	 * the harness.
+	 * @param type The type of event (e.g. `click` or `mousedown`)
+	 * @param options Options which can modify the event sent, like using a different EventClass or selecting a different
+	 *                        node to target, or provide the event initialisation properties
+	 */
+	public sendEvent<I extends EventInit>(type: string, options: HarnessSendEventOptions<I> = {}): this {
+		const { target = this.getDom() } = options;
+		sendEvent(target, type, options);
+		return this;
+	}
+
+	/**
+	 * Set the children that will be used when rendering the harnessed widget
+	 * @param children The children to be set on the harnessed widget
+	 */
+	public setChildren(...children: DNode[]): this {
```

Comment:
We sort of discussed this before... `.setChildren(...children)` is in my opinion more "usable" and we discussed that `WidgetBase.setChildren()` isn't actually meant to be a consumable API (and that we were considering renaming it `.__setChildren__()` to indicated that).  Given that people shouldn't be using it anyways, there is no need for them to conform, and if they don't need to conform, why not use the more usable API?  Is there really any value in a single array argument?  If you have an array of children, just spread them.  If not, you can avoid the literal array braces.